cmd/generate-catalog/generate.go

```diff
@@ -0,0 +1,227 @@
+package main
+
+import (
+	// needed for digest algorithm validation
+	_ "crypto/sha256"
+
+	"encoding/base64"
+	"fmt"
+	"log"
+	"os"
+	"slices"
+
+	semver "github.com/Masterminds/semver/v3"
+	"github.com/distribution/reference"
+	"github.com/goccy/go-yaml"
+)
+
+const (
+	inputFile                       = "bundles.yaml"
+	outputFile                      = "catalog-template.yaml"
+	iconFile                        = "icon.png"
+	firstLineHeadComment            = "DO NOT EDIT. This file is generated by cmd/generate-catalog/generate.go script."
+	secondLineHeadComment           = "Any manual changes will be overwritten."
+	channelDeprecationMessage       = "This version is no longer supported. Please switch to the `stable` channel or a channel for a version that is still supported.\n"
+	bundleDeprecationMessage        = "This operator bundle version is no longer supported. Please switch to non deprecated bundle version for support.\n"
+	deprecationMessageLatestChannel = "The `latest` channel is no longer supported.  Please switch to the `stable` channel.\n"
```

Comment:
It should rather be
```suggestion
	latestChannelDeprecationMessage = "The `latest` channel is no longer supported.  Please switch to the `stable` channel.\n"
```
for consistency with the constants above.